montepy/input_parser/data_parser.py

```diff
@@ -166,3 +166,107 @@ def data_classifier(self, p):
         return syntax_node.SyntaxNode(
             "data input classifier", {"start_pad": padding, "classifier": p.classifier}
         )
+
+
+class ParamOnlyDataParser(DataParser):
+    """
+    A parser for parsing parameter (key-value pair) only data inputs.
+
+    .e.g., SDEF
+
+    .. versionadded:: 0.2.11
+
+    :returns: a syntax tree for the data input.
+    :rtype: SyntaxNode
+    """
+
+    debugfile = None
+
+    @_(
+        "param_introduction spec_parameters",
+    )
+    def param_data_input(self, p):
+        ret = {}
+        for key, node in p.param_introduction.nodes.items():
+            ret[key] = node
+        if hasattr(p, "spec_parameters"):
+            ret["parameters"] = p.spec_parameters
+        return syntax_node.SyntaxNode("data", ret)
+
+    @_(
+        "classifier_phrase",
+        "padding classifier_phrase",
+    )
+    def param_introduction(self, p):
+        ret = {}
+        if isinstance(p[0], syntax_node.PaddingNode):
+            ret["start_pad"] = p[0]
+        else:
+            ret["start_pad"] = syntax_node.PaddingNode()
+        ret["classifier"] = p.classifier_phrase
+        ret["keyword"] = syntax_node.ValueNode(None, str, padding=None)
+        return syntax_node.SyntaxNode("data intro", ret)
+
+    @_("spec_parameter", "spec_parameters spec_parameter")
+    def spec_parameters(self, p):
+        """
+        A list of the parameters (key, value pairs) for this input.
+
+        :returns: all parameters
+        :rtype: ParametersNode
+        """
+        if len(p) == 1:
+            params = syntax_node.ParametersNode()
+            param = p[0]
+        else:
+            params = p[0]
+            param = p[1]
+        params.append(param)
```

Comment:
This looks weird; is it just a specific nested list flattener?

```python
>>> p = [ [a, b], d ]
>>> params = p[0]
[a, b]
>>> param = p[1]
d
>>> params.append(param)
[a, b, d]
```

Reply:
Yes you are exactly right in how you understand it.